sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -893,26 +901,84 @@ __SYCL_EXPORT bool is_source_kernel_bundle_supported(backend BE,
                                                      source_language Language);
 
 __SYCL_EXPORT kernel_bundle<bundle_state::ext_oneapi_source>
+make_kernel_bundle_from_source(
+    const context &SyclContext, source_language Language,
+    sycl::detail::string_view Source,
+    std::vector<std::pair<sycl::detail::string_view, sycl::detail::string_view>>
+        IncludePairsVec);
+
+__SYCL_EXPORT inline kernel_bundle<bundle_state::ext_oneapi_source>
 make_kernel_bundle_from_source(
     const context &SyclContext, source_language Language,
     const std::string &Source,
-    std::vector<std::pair<std::string, std::string>> IncludePairsVec);
+    std::vector<std::pair<std::string, std::string>> IncludePairsVec) {
+  size_t n = IncludePairsVec.size();
+  std::vector<std::pair<sycl::detail::string_view, sycl::detail::string_view>>
+      PairVec;
+  PairVec.reserve(n);
+  for (auto &Pair : IncludePairsVec)
+    PairVec.push_back({sycl::detail::string_view{Pair.first},
+                       sycl::detail::string_view{Pair.second}});
+
+  return make_kernel_bundle_from_source(
+      SyclContext, Language, sycl::detail::string_view{Source}, PairVec);
+}
 
 #if (!defined(_HAS_STD_BYTE) || _HAS_STD_BYTE != 0)
 __SYCL_EXPORT kernel_bundle<bundle_state::ext_oneapi_source>
 make_kernel_bundle_from_source(
     const context &SyclContext, source_language Language,
     const std::vector<std::byte> &Bytes,
-    std::vector<std::pair<std::string, std::string>> IncludePairsVec);
+    std::vector<std::pair<sycl::detail::string_view, sycl::detail::string_view>>
+        IncludePairsVec);
+
+__SYCL_EXPORT inline kernel_bundle<bundle_state::ext_oneapi_source>
```

Comment:
I don't believe we need to export these.
```suggestion
inline kernel_bundle<bundle_state::ext_oneapi_source>
```